packages/bloc/lib/src/bloc.dart

```diff
@@ -10,23 +9,24 @@ import '../bloc.dart';
 /// and transforms them into a `Stream` of `States` as output.
 /// {@endtemplate}
 abstract class Bloc<Event, State> extends Stream<State> implements Sink<Event> {
-  final PublishSubject<Event> _eventSubject = PublishSubject<Event>();
+  final _eventController = StreamController<Event>.broadcast();
+  final _stateController = StreamController<State>();
 
-  BehaviorSubject<State> _stateSubject;
+  State _currentState;
```

Comment:
nit but for consistency can we rename to `_state`?
```suggestion
  State _state;
```